src/scripts/federal-holidays-reminder.js

```diff
@@ -0,0 +1,88 @@
+const moment = require("moment-timezone");
+const scheduler = require("node-schedule");
+const holidays = require("@18f/us-federal-holidays");
+const {
+  slack: { postMessage },
+} = require("../utils");
+
+const CHANNEL = process.env.HOLIDAY_REMINDER_CHANNEL || "general";
+const TIMEZONE = process.env.HOLIDAY_REMINDER_TIMEZONE || "America/New_York";
+const reportingTime = moment(
+  process.env.HOLIDAY_REMINDER_TIME || "15:00",
+  "HH:mm"
+);
+const SUPPRESS_HERE = process.env.HOLIDAY_REMINDER_SUPPRESS_HERE;
+
+const suppressHere = (() => {
+  if (SUPPRESS_HERE) {
+    const upper = SUPPRESS_HERE.toUpperCase();
+    if (upper === "TRUE" || upper === "YES" || upper === "Y") {
+      return true;
+    }
+
+    if (!Number.isNaN(+SUPPRESS_HERE) && +SUPPRESS_HERE > 0) {
+      return true;
+    }
+  }
+  return false;
+})();
+
+const getNextHoliday = () => {
+  const now = moment.tz(TIMEZONE);
+
+  return holidays
+    .allForYear(now.year())
+    .concat(holidays.allForYear(now.year() + 1))
+    .map((h) => ({
+      ...h,
+      date: moment.tz(h.dateString, "YYYY-MM-DD", TIMEZONE),
+    }))
+    .filter((h) => h.date.isAfter(now))
+    .shift();
+};
+
+const previousWeekday = (date) => {
+  const source = moment(date);
+  source.subtract(1, "day");
+
+  let dow = source.format("dddd");
+  while (dow === "Saturday" || dow === "Sunday") {
+    source.subtract(1, "day");
+    dow = source.format("dddd");
+  }
+
+  return source;
+};
+
+const postReminder = (holiday) => {
+  postMessage({
+    channel: CHANNEL,
+    text: `${suppressHere ? "" : "@here "}Remember that *${holiday.date.format(
+      "dddd"
+    )}* is a federal holiday in observance of *${holiday.name}*!`,
+  });
+};
+
+const scheduleReminder = () => {
+  const nextHoliday = module.exports.getNextHoliday();
```

Comment:
Ah, so this is a trick so I can mock `getNextHoliday` in the tests. Without this, you end up mocking the module export, which turns into a different pointer, and so you can't actually prevent `scheduleReminder` from calling the real `getNextHoliday`. It's kind of ugly, but it simplifies the tests.

That said... this is another one where I don't like the tests. They are testing too much internal behavior instead of just testing the in/out. 😕